ensemble/volren/tests/test_volume_renderer.py

```diff
@@ -107,6 +115,23 @@ def test_volume_data_masking(self):
         points_with_mask = volume_data.render_data.number_of_points
         self.assertEqual(points_with_mask, mask_data.size)
 
+    def test_volume_data_filtering(self):
+        # Test apply filter on raw data, Pull Request #48
+
+        volume_data = self.viewer.volume_data
+
+        # Filter is not set initially
+        initial_points = volume_data.render_data.number_of_points
+        self.assertIsNone(volume_data.volume_filter)
+        # 256^3: that's because we are filtering the data before sending it
+        # to VTK, see `volume_data._resample_data`.
+        self.assertEqual(initial_points, 256 * 256 * 256)
+
+        # Now apply the filter
+        self.viewer.volume_data.volume_filter = SampleFilter(name='Sample')
+        filtered_points = volume_data.render_data.number_of_points
+        self.assertEqual(initial_points, filtered_points)
```

Comment:
This is a very weak test: you're not checking that the data has been filtered at all. This test would pass even if the data has not been re-computed at all!

Please select a combination of filter and data so that you know what to expect as a result, and test that the new `render_data` is equal to the expected value